src/materials/DamagePlasticityStressUpdate.C

```diff
@@ -0,0 +1,682 @@
+
+  params.addRangeCheckedParam<Real>("yield_strength_in_compression",
+                                    "yield_strength_in_compression >= 0",
+                                    "Absolute yield compressice strength");
+  params.addRequiredParam<Real>("compressive_damage_at_max_compressive_strength",
+                                "damage at maximum compressive strength");
+  params.addRequiredParam<Real>("maximum_strength_in_compression",
+                                "Absolute maximum compressive strength");
+  params.addRangeCheckedParam<Real>("fracture_energy_in_compression",
+                                    "fracture_energy_in_compression >= 0",
+                                    "Fracture energy of concrete in uniaxial compression");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "tip_smoother",
+      "tip_smoother>=0",
+      "Smoothing parameter: the cone vertex at mean = cohesion*cot(friction_angle), will be "
+      "smoothed by the given amount. Typical value is 0.1*cohesion");
+  params.addParam<bool>("perfect_guess",
+                        true,
+                        "Provide a guess to the Newton-Raphson proceedure "
+                        "that is the result from perfect plasticity.  With "
+                        "severe hardening/softening this may be "
+                        "suboptimal.");
+  params.addClassDescription("Damage Plasticity Model for concrete");
+  return params;
+}
+
+DamagePlasticityStressUpdate::DamagePlasticityStressUpdate(const InputParameters & parameters)
+  : MultiParameterPlasticityStressUpdate(parameters, 3, 1, 2),
+    _f_tol(getParam<Real>("yield_function_tol")),
+
+    _alfa(getParam<Real>("factor_relating_biaxial_unixial_cmp_str")),
+    _alfa_p(getParam<Real>("factor_controlling_dilatancy")),
+    _s0(getParam<Real>("stiff_recovery_factor")),
+
+    _Chi(getParam<Real>("ft_ep_slope_factor_at_zero_ep")),
+    _Dt(getParam<Real>("tensile_damage_at_half_tensile_strength")),
+    _ft(getParam<Real>("yield_strength_in_tension")),
+    _FEt(getParam<Real>("fracture_energy_in_tension")),
+
+    _fyc(getParam<Real>("yield_strength_in_compression")),
+    _Dc(getParam<Real>("compressive_damage_at_max_compressive_strength")),
+    _fc(getParam<Real>("maximum_strength_in_compression")),
+    _FEc(getParam<Real>("fracture_energy_in_compression")),
+
+    _at(1.5 * std::sqrt(1 - _Chi) - 0.5),
+    _ac((2. * (_fc / _fyc) - 1. + 2. * std::sqrt(std::pow((_fc / _fyc), 2.) - _fc / _fyc))),
+
+    _zt((1. + _at) / _at),
+    _zc((1. + _ac) / _ac),
+    _dPhit(_at * (2. + _at)),
+    _dPhic(_ac * (2. + _ac)),
+    _sqrtPhit_max((1. + _at + sqrt(1. + _at * _at)) / 2.),
+    _sqrtPhic_max((1. + _ac) / 2.),
+    _dt_bt(log(1. - _Dt) / log((1. + _at - sqrt(1. + _at * _at)) / (2. * _at))),
+    _dc_bc(log(1. - _Dc) / log((1. + _ac) / (2. * _ac))),
+    _ft0(0.5 * _ft /
+         ((1. - _Dt) * pow((_zt - _sqrtPhit_max / _at), (1. - _dt_bt)) * _sqrtPhit_max)),
```

Comment:
Change most instances of `pow` to `std::pow`

Reply:
Addressed